markdown.dtx

```diff
@@ -25871,35 +26274,128 @@ M.extensions.tex_math_dollars = function()
+
+      local display_math_opening_dollars  = parsers.dollar
+                                          * parsers.dollar
+
+      local display_math_closing_dollars  = parsers.dollar
+                                          * parsers.dollar
 
-      local inlinemath = between(C( parsers.backslash^-1 
-                                  * parsers.any
-                                  - parsers.blankline^2 
-                                  - parsers.dollar),
-                                 parsers.dollar * #(parsers.nonspacechar),
-                                 inlinemathtail)
+      local display_math_dollars = between(C( dollar_math_content),
+                                           display_math_opening_dollars,
+                                           display_math_closing_dollars)
 
-      local displaymathdelim  = parsers.dollar 
-                              * parsers.dollar
+      -- tex_math_?_backslash
+      local backslash_math_content  = parsers.any 
+                                    - parsers.blankline^2
 
-      local displaymath = between(C( parsers.backslash^-1 
-                                   * parsers.any 
-                                   - parsers.blankline^2 
-                                   - parsers.dollar),
-                                  displaymathdelim,
-                                  displaymathdelim)
+      -- tex_math_double_backslash
+      local inline_math_opening_double  = parsers.backslash
+                                        * parsers.backslash
+                                        * parsers.lparent
+                                        * #(parsers.nonspacechar)
 
-      local TexMathDollars = displaymath / writer.display_math
-                           + inlinemath / writer.inline_math
+      local inline_math_closing_double  = allowed_before_closing
+                                        * parsers.backslash
+                                        * parsers.backslash
+                                        * parsers.rparent
+
+      local inline_math_double  = between(C( backslash_math_content),
+                                          inline_math_opening_double,
+                                          inline_math_closing_double)
+
+      local display_math_opening_double = parsers.backslash
+                                        * parsers.backslash
+                                        * parsers.lbracket
+
+      local display_math_closing_double = allowed_before_closing
+                                        * parsers.backslash
+                                        * parsers.backslash
+                                        * parsers.rbracket
+
+      local display_math_double = between(C( backslash_math_content),
+                                          display_math_opening_double,
+                                          display_math_closing_double)
+
+      -- tex_math_single_backslash
```

Comment:
Instead of an opaque comment `-- tex_math_single_backslash`, how about a sentence or two in natural language?

```suggestion
%    \end{macrocode}
% \begin{markdown}
%
% The following patterns implement the Pandoc `tex_math_single_backslash` syntax extension.
%
% \end{markdown}
%  \begin{macrocode}
```